riscv_config/constants.py

```diff
@@ -6,19 +6,33 @@
 isa_schema = os.path.join(root, 'schemas/schema_isa.yaml')
 debug_schema = os.path.join(root, 'schemas/schema_debug.yaml')
 platform_schema = os.path.join(root, 'schemas/schema_platform.yaml')
+Zvl_extensions = [
+        "Zvl32b", "Zvl64b", "Zvl128b", "Zvl256b", "Zve512b", "Zvl1024b"
+]
+Zvef_extensions = [
+        "Zve32f", "Zve64f"
+]
+Zved_extensions = [
+        "Zve64d"
+]
+Zve_extensions = [
+        "Zve32x",  "Zve64x"
+] + Zvef_extensions + Zved_extensions
+
+Zb_extensions = ["Zba", "Zbb", "Zbc", "Zbe", "Zbf", "Zbm", "Zbp", "Zbr", "Zbs", "Zbt"]
+Zp_extensions = ["Zbpbo", "Zpn", "Zpsf"]
+
 Z_extensions = [
         "Zicbom", "Zicbop", "Zicboz", "Zicsr", "Zifencei", "Zihintpause",
         "Zam",
         "Zfh",
         "Zfinx", "Zdinx", "Zhinx", "Zhinxmin",
         "Ztso",
-        "Zba", "Zbb", "Zbc", "Zbe", "Zbf", "Zbkb", "Zbkc", "Zbkx", "Zbm", "Zbp", "Zbpbo", "Zbr", "Zbs", "Zbt",
+        "Zbkb", "Zbkc", "Zbkx",
         "Zk", "Zkn", "Zknd", "Zkne", "Zknh", "Zkr", "Zks", "Zksed", "Zksh", "Zkt",
-        "Zpn", "Zpsf",
         "Zmmul",
         "Svnapot"
-]
-Zb_extensions = ["Zba", "Zbb", "Zbc", "Zbe", "Zbf", "Zbm", "Zbp", "Zbr", "Zbs", "Zbt"]
-Zp_extensions = ["Zbpbo", "Zpn", "Zpsf"]
+] + Zvl_extensions + Zve_extensions + Zb_extensions + Zp_extensions
```

Comment:
This is wrong as this list should always be canonically ordered instead of just a list of Z extensions. The ordering requirements for the Z extensions are as follows:
`“Z” extensions are named, they should be ordered first by category, then alphabetically within a
category—for example, “Zicsr Zifencei Zam”`
Category ordering - `IMAFDQLCBKJTPV`
Further, the `Zmmul` extension should be moved before the `Zam` extension.

Reply:
Oh, thanks for pointing this out, I didn't know about this rule. But it seems that we don't check the order yet, so the order of the array doesn't really matter, at least for now. Do we want to add this check?

Reply:
The order isn't explicitly checked because the way the regex is constructed enforces this order on the input string. Hence the need for the list to be ordered in the correct way.

Reply:
AFAIK, regexp doesn't guarantee any order, e.g. `(a|b)*` can match both `ab` and `ba`.

Reply:
My bad. The older version enforced ordering using regex. The newer version does this programmatically in `isa_validator.py`. However for the Z extensions the ordering in this list is considered as canonical.

Reply:
But it's right keeping them in the correct order. I have reverted the previous changes and corrected the order of Zmmul, Zve_extensions, and Zvl_extensions, force-pushed.